dcpy/connectors/edm/recipes.py

```diff
@@ -60,7 +62,8 @@ def assign_file_type(self, file_type_preferences: list[DatasetType]) -> DatasetT
             raise FileNotFoundError(
                 f"No datasets of types {file_type_preferences} found in {self.s3_folder}"
             )
-        self.file_type = file_type
+        if mutate:
+            self.file_type = file_type
```

Comment:
Wanted this because in the case of something like reading a dataframe rather than flushing out a recipe, it makes sense to just get the file type rather than mutate an object